client.go

```diff
@@ -517,22 +517,66 @@ func (c *Client) Whisper(username, text string) {
 	c.send(fmt.Sprintf("PRIVMSG #%s :/w %s %s", c.ircUser, username, text))
 }
 
-// Join enter a twitch channel to read more messages
-func (c *Client) Join(channel string) {
-	channel = strings.ToLower(channel)
-
-	// If we don't have the channel in our map AND we have an
-	// active connection, explicitly join before we add it to our map
+// Join enter a twitch channel to read more messages.
+// Any channels which could not be joined due to the maximum
+// message length being reached are returned in a slice.
+func (c *Client) Join(channels ...string) []string {
+	message, joined, remainder := createJoinMessage(c.channels, channels...)
+
+	// If we have an active connection, explicitly join
+	// before we add the joined channels to our map
 	c.channelsMtx.Lock()
-	if !c.channels[channel] && c.connActive.get() {
-		go c.send(fmt.Sprintf("JOIN #%s", channel))
+	if c.connActive.get() {
+		go c.send(message)
 	}
 
-	c.channels[channel] = true
-	c.channelUserlistMutex.Lock()
-	c.channelUserlist[channel] = map[string]bool{}
-	c.channelUserlistMutex.Unlock()
+	for _, channel := range joined {
+		c.channels[channel] = true
+		c.channelUserlistMutex.Lock()
+		c.channelUserlist[channel] = map[string]bool{}
+		c.channelUserlistMutex.Unlock()
+	}
 	c.channelsMtx.Unlock()
+
+	return remainder
+}
+
+var maxMessageLength = 510
```

Comment:
please move this to the IRC file and make a constant out of it. It's not variable.

Reply:
Moved to IRC file and made constant